src/compiler/checker.ts

```diff
@@ -24587,36 +24587,26 @@ namespace ts {
         }
 
         function getRootSymbols(symbol: Symbol): Symbol[] {
+            const roots = getImmediateRootSymbols(symbol);
+            return roots === undefined ? [symbol] : flatMap(roots, getRootSymbols);

```

Comment:
nit: This would read better flipped since I think that's the common (or at least more important) case. And since only `undefined` is `falsy` in the type of `roots`, you can just say `roots ? flatMap...`

Reply:
I was going to ask for a performance analysis of this change since it does more work and conses more, but I'm pretty sure `getRootSymbols` is only called interactively, and only in findAllReferences, once per textual match of the requested referent. Is this correct?

Reply:
Right, it's only used in `findAllReferences`. It's called on every match though. But usually there is nothing to do and it returns a single-element array just like it did before.